src/naemon/workers.c

```diff
@@ -708,6 +711,31 @@ int init_workers(int desired_workers)
 	if (desired_workers < (int)workers.len)
 		return -1;
 
+	return desired_workers;
```

Comment:
since this function now returns the number of desired workers, it should always do this. Previously this function had some special return values (0/-1) which should be replaced as well to return the number of desired workers. (ex.: line 708/712).
I'd say this function should always simply return the desired workers and conclusions should be left for the caller sub.

Reply:
It definitely makes sense.

Reply:
Thanks. I have pulled your changes into this PR